### PR TITLE
kubelet: Small refactor on GetPhase()

### PR DESCRIPTION
**Pull request by yifan-gu**
cc @dchen1107 @yujuhong @Random-Liu @timstclair

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -2955,29 +2955,33 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
...
+			running++
+		case containerStatus.State.Terminated != nil:
+			stopped++
+			if containerStatus.State.Terminated.ExitCode == 0 {
+				succeeded++
+			} else {
+				failed++
+			}
+		case containerStatus.State.Waiting != nil:
+			if containerStatus.LastTerminationState.Terminated != nil {
```
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: The definition of `stopped` is unclear at best. I think #17971 also needs to be clearly defined...

Comment (Random-Liu) [inline pkg/kubelet/kubelet.go]: Yeah...I'm totally confused...Why the waiting container should be calculated as stopped here? @yujuhong

Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: Because they are in the crashloop backoff state :-(

Comment (Random-Liu) [inline pkg/kubelet/kubelet.go]: Ah...I didn't read the following code just now...Thanks! :)

**Comment by Random-Liu** [comment]
@yifan-gu LGTM. Switch really is cleaner than if-else chain (I just took a Golang class today, the teacher also told us this, hahaha)

**Comment by yifan-gu** [comment]
@k8s-bot test this

**Comment by yifan-gu** [comment]
@Random-Liu Thanks for fixing the title :) I fixed the commit title as well.

**Pull request merged**